tests/products/test_file.py

```diff
@@ -301,25 +301,28 @@ def test_task_without_client_is_outdated_by_code(tmp_directory):
     assert t.product._is_outdated()
 
 
-def _edit_source_code(path):
+def _edit_source_code(path, tmp_path):
+    tmp_path = tmp_path / "temp.txt"
     m = _load_json(path)
     m['stored_source_code'] = m['stored_source_code'] + '1 + 1\n'
-    _write_json(m, path)
+    with open(tmp_path, 'w') as f:
+        json.dump(m, f)
+    os.replace(tmp_path, path)
 
 
-def _delete_metadata(path):
+def _delete_metadata(path, tmp_path):
```

Comment:
doesn't look like tmp_path is used here

Reply:
Ya but I had to pass it because we have a common `operation` call where `operation` can be `_delete` or `_edit_source_code`. The tmp_path is being required in `_edit_source_code`